Source/UI/Controllers/ThreedsWebViewController.swift

```diff
@@ -36,18 +41,52 @@ public class ThreedsWebViewController: UIViewController {
         let successUrl = URL(string: successUrlString)
         let failUrl = URL(string: failUrlString)
 
-        self.init(successUrl: successUrl, failUrl: failUrl, urlHelper: URLHelper())
+        self.init(successUrl: successUrl, failUrl: failUrl)
+    }
+
+    public convenience init (successUrl: URL, failUrl: URL) {
+        self.init(successUrl: .some(successUrl), failUrl: .some(failUrl))
     }
 
     /// Initializes a web view controller adapted to handle 3dsecure.
-    public convenience init(successUrl: URL, failUrl: URL) {
-        self.init(successUrl: successUrl, failUrl: failUrl, urlHelper: URLHelper())
+    convenience init(successUrl: URL?, failUrl: URL?) {
+        // in 4.0.0 release we should ask for the environment
```

Comment:
Is this ToDo ?

Reply:
kind of? we don't want to change the public API now but it would be good to have the correct environment against the logs